core/parser/src/parser/expression/assignment/mod.rs

```diff
@@ -151,11 +154,14 @@ where
         let peek_token = cursor.peek(0, interner).or_abrupt()?;
         let position = peek_token.span().start();
         let start_linear_span = peek_token.linear_span();
-        let mut lhs = ConditionalExpression::new(self.allow_in, self.allow_yield, self.allow_await)
+        let lhs = ConditionalExpression::new(self.allow_in, self.allow_yield, self.allow_await)
             .parse(cursor, interner)?;
 
         // If the left hand side is a parameter list, we must parse an arrow function.
-        if let Expression::FormalParameterList(parameters) = lhs {
+        if let FormalParameterListOrExpression::FormalParameterList {
+            fpl: parameters, ..
+        } = lhs
+        {
```

Comment:
Remark: Could match here, for both variants (`FormalParameterList`, and `Expression`) to avoid the `expect_expression()` below.

Something like:

```rust
let mut lhs = match lhs {
    FormalParameterListOrExpression::FormalParameterList { .. } => {
        /* ... */
    }
    FormalParameterListOrExpression::Expression(lhs) => lhs,
};
```